helper/max_length_slice.go

```diff
@@ -0,0 +1,64 @@
+package helper
+
+import "sync"
+
+// MaxLengthStringSliceAdaptor wrap the given slice with several methods for MaxLenSlice.
+// Note that this adaptor does not own that slice at all, so keep the content of slice unchanged!
+type MaxLengthStringSliceAdaptor struct {
+	s      []string
+	maxlen int
+	lock   sync.Mutex
```

Comment:
Use a `RWMutex` here to allow reading in parallel or use a channel to sync the writing actions.